ext/IntegralsZygoteExt.jl

```diff
@@ -76,26 +93,39 @@ function ChainRulesCore.rrule(::typeof(Integrals.__solvebp), cache, alg, senseal
             do_inf_transformation = Val(false),
             cache.kwargs...)
 
-        if p isa Number
-            dp = Integrals.__solvebp_call(dp_cache, alg, sensealg, lb, ub, p; kwargs...)[1]
-        else
-            dp = Integrals.__solvebp_call(dp_cache, alg, sensealg, lb, ub, p; kwargs...).u
-        end
+        project_p = ProjectTo(p)
+        dp = project_p(Integrals.__solvebp_call(dp_cache,
+            alg,
+            sensealg,
+            lb,
+            ub,
+            p;
+            kwargs...).u)
 
         if lb isa Number
-            dlb = -_f(lb)
-            dub = _f(ub)
+            dlb = cache.batch > 0 ? -_f([lb]) : -_f(lb)
+            dub = cache.batch > 0 ? _f([ub]) : _f(ub)
             return (NoTangent(), NoTangent(), NoTangent(), NoTangent(), dlb, dub, dp)
         else
+            # we need to compute 2*length(lb) integrals on the faces of the hypercube, as we
+            # can see from writing the multidimensional integral as an iterated integral
+            # alternatively we can use Stokes' theorem to replace the integral on the
+            # boundary with a volume integral of the flux of the integrand
+            # ∫∂Ω ω = ∫Ω dω, which would be better since we won't have to change the
+            # dimensionality of the integral or the quadrature used (such as quadratures
+            # that don't evaluate points on the boundaries) and it could be generalized to
+            # other kinds of domains. The only question is to determine ω in terms of f and
+            # the deformation of the surface (e.g. consider integral over an ellipse and
+            # asking for the derivative of the result w.r.t. the semiaxes of the ellipse)
             return (NoTangent(), NoTangent(), NoTangent(), NoTangent(), NoTangent(),
                 NoTangent(), dp)
         end
     end
     out, quadrature_adjoint
 end
 
-Zygote.@adjoint function Zygote.literal_getproperty(sol::SciMLBase.IntegralSolution,
-    ::Val{:u})
-    sol.u, Δ -> (SciMLBase.build_solution(sol.prob, sol.alg, Δ, sol.resid),)
-end
+# Zygote.@adjoint function Zygote.literal_getproperty(sol::SciMLBase.IntegralSolution,
+#     ::Val{:u})
+#     sol.u, Δ -> (SciMLBase.build_solution(sol.prob, sol.alg, Δ, sol.resid),)
+# end
```

Comment:
Why was this removed?

Reply:
That just means a test didn't use `sol.u`. This is necessary for if the user getfield's on `u`. It should be added back for that case, preferably with a test since that seems to have been missed.